kura/org.eclipse.kura.core.configuration/src/main/java/org/eclipse/kura/core/configuration/ConfigurationServiceImpl.java

```diff
@@ -706,20 +706,40 @@ boolean mergeWithDefaults(OCD ocd, Map<String, Object> properties) throws KuraEx
         return ComponentUtil.getDefaultProperties(ocd, this.ctx);
     }
 
-    Map<String, Object> decryptPasswords(ComponentConfiguration config) {
+    Map<String, Object> decryptConfigurationProperties(ComponentConfiguration config) {
         Map<String, Object> configProperties = config.getConfigurationProperties();
         for (Entry<String, Object> property : configProperties.entrySet()) {
-            if (property.getValue() instanceof Password) {
+            Object configValue = property.getValue();
+            
+            if (configValue instanceof Password || configValue instanceof Password[]) {
                 try {
-                    Password decryptedPassword = new Password(
-                            this.cryptoService.decryptAes(property.getValue().toString().toCharArray()));
-                    configProperties.put(property.getKey(), decryptedPassword);
-                } catch (Exception e) {
+                    Object decryptedValue = decryptPasswordProperties(configValue);
+                    configProperties.put(property.getKey(), decryptedValue);
+                } catch (Exception e) {       
                 }
             }
         }
         return configProperties;
     }
+    
+    private Object decryptPasswordProperties(Object encryptedValue) throws KuraException{
```

Comment:
I believe it would be better to have an Password[] decryptPasswordProperties(Password[] encryptedValue) method and avoid the type checks that the caller has already done.